pkg/liquidity-mining/contracts/gauges/StakelessGaugeController.sol

```diff
@@ -0,0 +1,223 @@
+        override
+        gaugeTypeChecked(gaugeType)
+    {
+        EnumerableSet.AddressSet storage gaugesForType = _gauges[gaugeType];
+
+        for (uint256 i = 0; i < gauges.length; i++) {
+            IStakelessGauge gauge = gauges[i];
+            require(
+                _gaugeAdder.isGaugeFromValidFactory(address(gauge), gaugeType),
+                "Gauge does not come from valid factory"
+            );
+            require(_gaugeController.gauge_exists(address(gauge)), "Gauge does not exist in controller");
+            require(!gauge.is_killed(), "Gauge was killed");
+            require(gaugesForType.add(address(gauge)), "Gauge already present");
+        }
+        emit IStakelessGaugeController.GaugesAdded(gaugeType, gauges);
+    }
+
+    /**
+     * @dev See IStakelessGaugeController#removeGauges.
+     */
+    function removeGauges(IGaugeAdder.GaugeType gaugeType, IStakelessGauge[] calldata gauges)
+        external
+        override
+        gaugeTypeChecked(gaugeType)
+    {
+        EnumerableSet.AddressSet storage gaugesForType = _gauges[gaugeType];
+
+        for (uint256 i = 0; i < gauges.length; i++) {
+            IStakelessGauge gauge = gauges[i];
+            require(gauge.is_killed(), "Gauge was not killed");
+            require(gaugesForType.remove(address(gauge)), "Gauge not present");
+        }
+        emit IStakelessGaugeController.GaugesRemoved(gaugeType, gauges);
+    }
+
+    /**
+     * @dev See IStakelessGaugeController#hasGauge.
+     */
+    function hasGauge(IGaugeAdder.GaugeType gaugeType, IStakelessGauge gauge)
+        external
+        view
+        override
+        gaugeTypeChecked(gaugeType)
+        returns (bool)
+    {
+        return _gauges[gaugeType].contains(address(gauge));
+    }
+
+    /**
+     * @dev See IStakelessGaugeController#getTotalGauges.
+     */
+    function getTotalGauges(IGaugeAdder.GaugeType gaugeType)
+        external
+        view
+        override
+        gaugeTypeChecked(gaugeType)
+        returns (uint256)
+    {
+        return _gauges[gaugeType].length();
+    }
+
+    /**
+     * @dev See IStakelessGaugeController#getGaugeAt.
+     */
+    function getGaugeAt(IGaugeAdder.GaugeType gaugeType, uint256 index)
+        external
+        view
+        override
+        gaugeTypeChecked(gaugeType)
+        returns (address)
+    {
+        return _gauges[gaugeType].at(index);
+    }
+
+    /**
+     * @dev See IStakelessGaugeController#checkpointGaugesAboveRelativeWeight.
+     * Unspent ETH is sent back to sender.
+     */
+    function checkpointGaugesAboveRelativeWeight(uint256 minRelativeWeight) external payable override nonReentrant {
+        (uint256 singleArbGaugeBridgeETH, uint256 totalArbBridgeETH) = _getArbitrumBridgeCosts();
+
+        // solhint-disable-next-line not-rely-on-time
+        uint256 currentPeriod = _roundDownTimestamp(block.timestamp);
+
+        _checkpointGauges(IGaugeAdder.GaugeType.Polygon, minRelativeWeight, currentPeriod, 0);
```

Comment:
I'd consider computing the value _inside_ `_checkpointGauges` instead of here (i.e. 0 if non arbitrum, whatever the bridge returns otherwise). That'll simplify some things, such as making it obvious why it is correct to return 0 in the current `getBridgeCosts()` impl if we have no arbitrum gauges (which is right now a bit indirect).

Reply:
Done.